### PR TITLE
SO library via autotools machinery

### PR DESCRIPTION
**Pull request by jgmbenoit**
In view to build a SO library  for plfit in such a way it can be use in our own code, this chain of patches sets up a parallel autotools machinery. For now, the Python part is not set up., but it is a matter of time. Pleas ehave a look and let me know what you think.

### CODE REVIEW

File: src/plfit.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #define __PLFIT_H__
 
 #include <stdlib.h>
-#include "plfit_mt.h"
```
Comment (ntamas) [inline src/plfit.h]: Why is this change necessary? Wouldn't it work simply with the original double-quotes?

**Comment by ntamas** [comment]
Commit `f109b2a` could be merged into the trunk as is; I would be happy to merge it if you separate it into a new PR.

**Comment by ntamas** [comment]
Re `f8c8474`: does `hzeta.c` use anything from `config.h` so that it needs to depend on it?

**Comment by ntamas** [comment]
Commit `cfee8d5` is a good idea; I think it could be merged as is (if you file a separate PR with it).

**Comment by ntamas** [comment]
In general: I appreciate all the efforts you have put into implementing an `autotools` machinery around `plfit` very much; however, I am a bit reluctant to include it in master because my preferred toolset is CMake and I cannot guarantee that I won't break anything with the `autotools` part - and I will probably not notice it if I do because I won't be using it regularly. So, even if I merge it in master, I cannot take the responsibility of maintaining it if anything breaks.

May I ask you what is your overall goal with adding an `autotools`-based build system? Is it only to be able to build a shared library? Or is there another project that would integrate `plfit` and that cannot cope with the build files that CMake generates?

**Comment by jgmbenoit** [comment]
My motivation was to package plfit for Debian (I am a DD). Now I can see that plfit comes with igraph.

**Comment by ntamas** [comment]
CMake will build a shared library if you configure it like this:

```
mkdir build
cd build
cmake .. -DBUILD_SHARED_LIBS=on
```

I haven't actually _tried_ the shared libs, but they seem OK for me. One problem with CMake is that you cannot build both shared _and_ static libs in the same compilation pass, you need to configure CMake once for static libs and once for shared.

**Comment by jgmbenoit** [comment]
I can build in two passes the shared and the static libraries. However, they are not getting install at install time. Must we add some cmake material ?

**Comment by jgmbenoit** [comment]
I have just figured out how to install the shared and static libraries with cmake.

**Comment by jgmbenoit** [comment]
I am on my way to pull similar material but cmake centric ones.
